Data/HashMap/Base.hs

```diff
@@ -2065,6 +2186,15 @@ updateOrConcatWithKey f ary1 ary2 = A.run $ do
     return mary
 {-# INLINABLE updateOrConcatWithKey #-}
 
+-- | /O(n*m)/ Check if the first array is a subset of the second array.
+subsetArray :: Eq k => (v1 -> v2 -> Bool) -> A.Array (Leaf k v1) -> A.Array (Leaf k v2) -> Bool
+subsetArray cmpV ary1 ary2 = A.length ary1 <= A.length ary2 && go 0 (A.length ary1)
+  where
+    go !i n
+      | i >= n = True
+      | otherwise = let (L k1 v1) = A.index ary1 i
+                    in lookupInArrayCont (\_ -> False) (\v2 _ -> cmpV v1 v2 && go (i+1) n) k1 ary2
```

Comment:
The `lookupInArrayCont`-expression is quite a mouthful! ;)

Is this complexity "essential"? Or could this be implemented more readably, for example with a `foldr` over `ary1`?

I think we could also exploit the fact that once we have found a corresponding element at a certain index in `ary2`, we don't have to check that index again. Does that sound worthwhile?

Reply:
I changed the implementation to use `Array.all`. It looks a bit more readable now.